flytekit/clis/sdk_in_container/register.py

```diff
@@ -0,0 +1,161 @@
+    domain: str,
+    image_config: ImageConfig,
+    output: str,
+    destination_dir: str,
+    service_account: str,
+    raw_data_prefix: str,
+    version: typing.Optional[str],
+    package_or_module: typing.Tuple[str],
+):
+    """
+    This command produces a Flyte backend registrable package of all entities in Flyte.
+    For tasks, one pb file is produced for each task, representing one TaskTemplate object.
+    For workflows, one pb file is produced for each workflow, representing a WorkflowClosure object.  The closure
+        object contains the WorkflowTemplate, along with the relevant tasks for that workflow.
+        This serialization step will set the name of the tasks to the fully qualified name of the task function.
+    """
+
+    pkgs = ctx.obj[constants.CTX_PACKAGES]
+    if not pkgs:
+        print("No pkgs")
+    if pkgs:
+        raise ValueError("to do, please implement")
+
+    cli_logger.warning(
+        f"Running pyflyte register from {os.getcwd()} "
+        f"with images {image_config} "
+        f"and image destinationfolder {destination_dir} "
+        f"on {len(package_or_module)} package(s) {package_or_module}"
+    )
+
+    # Create and save FlyteRemote,
+    remote = get_and_save_remote_with_click_context(ctx, project, domain)
+
+    # Todo: add switch for non-fast - skip the zipping and uploading and no fastserializationsettings
+    # Create a zip file containing all the entries.
+    detected_root = find_common_root(package_or_module)
+    cli_logger.warning(f"Using {detected_root} as root folder for project")
+    zip_file = fast_package(detected_root, output)
+
+    # Upload zip file to Admin using FlyteRemote.
+    md5_bytes, native_url = remote._upload_file(pathlib.Path(zip_file))
+    cli_logger.warning(f"Uploaded zip {zip_file} to {native_url}")
+
+    # Create serialization settings
+    # Todo: Rely on default Python interpreter for now, this will break custom Spark containers
+    serialization_settings = SerializationSettings(
+        project=project,
+        domain=domain,
+        image_config=image_config,
+        fast_serialization_settings=FastSerializationSettings(
+            enabled=True,
+            destination_dir=destination_dir,
+            distribution_location=native_url,
+        ),
+    )
+
+    options = Options.default_from(k8s_service_account=service_account, raw_data_prefix=raw_data_prefix)
+
+    # Load all the entities
+    registerable_entities = load_packages_and_modules(
+        serialization_settings, detected_root, list(package_or_module), options
+    )
+    cli_logger.warning(f"Found and serialized {len(registerable_entities)} entities")
+
+    if not version:
+        version = remote._version_from_hash(md5_bytes, serialization_settings, service_account, raw_data_prefix)  # noqa
+        cli_logger.warning(f"Computed version is {version}")
+
+    # Register using repo code
+    repo_register(registerable_entities, project, domain, version, remote.client)
```

Comment:
is this the same as package? I hope so right

Reply:
yes i'm importing it from there